.pre-commit-hooks.yaml

```
@@ -0,0 +1,24 @@
+- id: poetry-check
+  name: poetry-check
+  description: run poetry check to validate config
+  entry: poetry check
+  language: python
+  language_version: python3
+  pass_filenames: false
+
+- id: poetry-lock
+  name: poetry-lock
+  description: run poetry lock to update lock file
+  entry: poetry lock
+  language: python
+  language_version: python3
+  pass_filenames: false
```

I think it makes sense to default to "no-update" here - otherwise running this on CI for the same commit will break when new packages are uploaded to pypi.org

```suggestion
  pass_filenames: false
  args: ["--no-update"]
```

---

I run `pre-commit` via GH Actions and also use `pre-commit.ci` and I do not think this is a good idea.

- You run `pre-commit` much more often locally than in CI. Therefore I prefer defaults for this case.
- My intention for this hook is to keep the lock file up-to-date and `--no-update` partially works against that.
- Also it does not happen that often, that a package is released while `pre-commit` is running in CI. When the run fails you can update the dependency accordingly and push again to trigger a new run. Or wait for e.g. `dependabot`.

Workarounds:
- You can run `SKIP=poetry-lock pre-commit run --all-files` and `poetry lock --no-update` in CI.
- `pre-commit.ci` can not only lint/check the files, but also commit changes from e.g. a formater back to a PR.